### ISSUE TITLE
Create one component to render all shelves

### ISSUE THREAD

**Issue by RivaD2** [Project-Books/book-project#762]
**Describe the solution you'd like**

Currently we have `ShelfBook` which is a single component. On the `myBooks` page, this shelf component is being rendered for each shelf.

**Task:**
  - Create one component that renders all the shelves (see `bookList.tsx`). The component should receive an array of books like `bookList.tsx` (see listview by toggling switch at bottom of `myBooks` page to list view), and dynamically populate them.

**Additional context**

**Shelf View seen here:**

<img width="1392" alt="myBooks-page" src="https://user-images.githubusercontent.com/67027945/135005780-5ce3210e-6371-4544-8f6c-46e30978bd1e.png">

**Note:** 
- Branch from the main branch and set main as the destination branch in your pull request
- Connect to the get all books endpoint on the backend (see `endpoints.ts` in `shared` folder for frontend)

**Comment by RivaD2** [comment]
@knjk04 can you confirm the endpoint needed for this one?

**Comment by knjk04** [comment]
@RivaD2 Ah, we'll need a new endpoint for this. I'll make a new ticket for that. Should we block this until that endpoint is ready?

**Comment by knjk04** [comment]
Blocked by #769

**Comment by knjk04** [comment]
Unblocked. GET endpoints (see https://github.com/Project-Books/book-project/blob/main/backend/src/main/java/com/karankumar/bookproject/backend/controller/PredefinedShelfController.java):
- /api/shelf/books/to-read
- /api/shelf/books/reading
- /api/shelf/books/read
- /api/shelf/books/did-not-finish

**Comment by varadeth** [comment]
Hi @knjk04 , I want to contribute in this enhancement. I want to get better understanding in this and then I will start working on this
